File: data/transcoder_evaluation_gfg/go/SMALLEST_SUM_CONTIGUOUS_SUBARRAY.go
package main

import (
		"fmt"
	"math"
	"os"
	"unsafe"
)

func min(x int, y int) int {
	if x < y {
		return x
	}
	return y
}
func max(x int, y int) int {
	if x > y {
		return x
	}
	return y
}
func cmpfunc(a unsafe.Pointer, b unsafe.Pointer) int {
	return *(*int)(a) - *(*int)(b)
}


func f_gold(arr []int, n int) int {
	var (
		min_ending_here int = math.MaxInt64
		min_so_far      int = math.MaxInt64
	)
	for i := int(0); i < n; i++ {
		if min_ending_here > 0 {
			min_ending_here = arr[i]
		} else {
			min_ending_here += arr[i]
		}
		min_so_far = min(min_so_far, min_ending_here)
	}
	return min_so_far
}
//TOFILL
func main() {
	var (
		n_success int      = 0
		param0_0  []int  = []int{2, 9, 13, 14, 15, 18, 19, 19, 25, 26, 29, 29, 29, 30, 31, 36, 37, 37, 38, 39, 39, 40, 40, 42, 42, 46, 50, 53, 58, 60, 62, 64, 65, 67, 68, 69, 72, 77, 78, 83, 85, 89, 90, 93, 95, 95, 97}
		param0_1  []int  = []int{14, -58, 8, 78, -26, -20, -60, 42, -64, -12}
		param0_2  []int  = []int{0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1}
		param0_3  []int  = []int{44, 88, 20, 47, 69, 42, 26, 49, 71, 91, 18, 95, 9, 66, 60, 37, 47, 29, 98, 63, 15, 9, 80, 66, 1, 9, 57, 56, 20, 2, 1}
		param0_4  []int  = []int{-78, -64, -62, -60, -52, 4, 8, 46, 72, 74}
		param0_5  []int  = []int{0, 0, 1, 0, 0, 0, 0, 1, 1, 0, 0, 1, 1, 0, 1, 1}
		param0_6  []int  = []int{3, 7, 16, 17, 23, 23, 23, 28, 29, 30, 34, 38, 40, 41, 43, 43, 44, 46, 51, 51, 51, 55, 57, 58, 61, 62, 66, 66, 67, 69, 70, 73, 75, 77, 79, 80, 85, 85, 87, 87, 93, 96}
		param0_7  []int  = []int{80, 22, 38, 26, 62, -48, -48, 46, -54, 4, 76, 46, 48, 40, -92, -98, -88, 12, -42, -94, 76, -16, -82, 62, 18, -24}
		param0_8  []int  = []int{0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1}
		param0_9  []int  = []int{85, 44, 1, 97, 50, 74, 62, 90, 3, 78, 8, 48, 96, 41, 36, 91, 57, 97, 85, 90, 78, 43, 28, 92, 85, 59, 29, 38, 34, 65, 20, 26, 27, 23, 71, 22, 47, 99, 68, 93, 67, 66, 69, 82, 98, 15, 44, 51, 65}
		param0    [][]int = [][]int{param0_0, param0_1, param0_2, param0_3, param0_4, param0_5, param0_6, param0_7, param0_8, param0_9}
		param1    []int  = []int{24, 6, 8, 26, 8, 11, 38, 22, 13, 45}
	)
	for i := int(0); i < len(param0[:]); i++ {
		if f_filled(([]int)(param0[i]), param1[i]) == f_gold(([]int)(param0[i]), param1[i]) {
			n_success += 1
		}
	}
	fmt.Print("#Results:", " ", n_success, ", ", len(param0[:]))
	os.Exit(0)
}

File: data/transcoder_evaluation_gfg/rust/SMALLEST_SUM_CONTIGUOUS_SUBARRAY.rs
#![feature(main)]
// Copyright (c) 2019-present, Facebook, Inc.
// All rights reserved.
//
// This source code is licensed under the license found in the
// LICENSE file in the root directory of this source tree.
//

#[no_mangle]
pub unsafe fn abs(mut x: i32)
 -> i32 {
    return x.abs();
}

#[no_mangle]
pub unsafe fn min(mut x: i32, mut y: i32)
 -> i32 {
    return if x < y { x } else { y };
}
#[no_mangle]
pub unsafe fn max(mut x: i32, mut y: i32)
 -> i32 {
    return if x > y { x } else { y };
}

pub unsafe fn f_gold(mut arr: *mut i32, mut n: i32)
 -> i32 {
    let mut min_ending_here: i32 = 2147483647 as i32;
    let mut min_so_far: i32 = 2147483647 as i32;
    let mut i: i32 = 0 as i32;
    while i < n {
        if min_ending_here > 0 as i32 {
            min_ending_here = *arr.offset(i as isize)
        } else { min_ending_here += *arr.offset(i as isize) }
        min_so_far = min(min_so_far, min_ending_here);
        i += 1
    }
    return min_so_far;
}
//TOFILL
unsafe fn main_0() -> i32 {
    let mut n_success: i32 = 0 as i32;
    let mut param0_0: [i32; 47] =
        [2 as i32, 9 as i32, 13 as i32,
         14 as i32, 15 as i32, 18 as i32,
         19 as i32, 19 as i32, 25 as i32,
         26 as i32, 29 as i32, 29 as i32,
         29 as i32, 30 as i32, 31 as i32,
         36 as i32, 37 as i32, 37 as i32,
         38 as i32, 39 as i32, 39 as i32,
         40 as i32, 40 as i32, 42 as i32,
         42 as i32, 46 as i32, 50 as i32,
         53 as i32, 58 as i32, 60 as i32,
         62 as i32, 64 as i32, 65 as i32,
         67 as i32, 68 as i32, 69 as i32,
         72 as i32, 77 as i32, 78 as i32,
         83 as i32, 85 as i32, 89 as i32,
         90 as i32, 93 as i32, 95 as i32,
         95 as i32, 97 as i32];
    let mut param0_1: [i32; 10] =
        [14 as i32, -(58 as i32), 8 as i32,
         78 as i32, -(26 as i32), -(20 as i32),
         -(60 as i32), 42 as i32, -(64 as i32),
         -(12 as i32)];
    let mut param0_2: [i32; 17] =
        [0 as i32, 0 as i32, 0 as i32,
         0 as i32, 0 as i32, 0 as i32,
         0 as i32, 0 as i32, 1 as i32,
         1 as i32, 1 as i32, 1 as i32,
         1 as i32, 1 as i32, 1 as i32,
         1 as i32, 1 as i32];
    let mut param0_3: [i32; 31] =
        [44 as i32, 88 as i32, 20 as i32,
         47 as i32, 69 as i32, 42 as i32,
         26 as i32, 49 as i32, 71 as i32,
         91 as i32, 18 as i32, 95 as i32,
         9 as i32, 66 as i32, 60 as i32,
         37 as i32, 47 as i32, 29 as i32,
         98 as i32, 63 as i32, 15 as i32,
         9 as i32, 80 as i32, 66 as i32,
         1 as i32, 9 as i32, 57 as i32,
         56 as i32, 20 as i32, 2 as i32,
         1 as i32];
    let mut param0_4: [i32; 10] =
        [-(78 as i32), -(64 as i32), -(62 as i32),
         -(60 as i32), -(52 as i32), 4 as i32,
         8 as i32, 46 as i32, 72 as i32,
         74 as i32];
    let mut param0_5: [i32; 16] =
        [0 as i32, 0 as i32, 1 as i32,
         0 as i32, 0 as i32, 0 as i32,
         0 as i32, 1 as i32, 1 as i32,
         0 as i32, 0 as i32, 1 as i32,
         1 as i32, 0 as i32, 1 as i32,
         1 as i32];
    let mut param0_6: [i32; 42] =
        [3 as i32, 7 as i32, 16 as i32,
         17 as i32, 23 as i32, 23 as i32,
         23 as i32, 28 as i32, 29 as i32,
         30 as i32, 34 as i32, 38 as i32,
         40 as i32, 41 as i32, 43 as i32,
         43 as i32, 44 as i32, 46 as i32,
         51 as i32, 51 as i32, 51 as i32,
         55 as i32, 57 as i32, 58 as i32,
         61 as i32, 62 as i32, 66 as i32,
         66 as i32, 67 as i32, 69 as i32,
         70 as i32, 73 as i32, 75 as i32,
         77 as i32, 79 as i32, 80 as i32,
         85 as i32, 85 as i32, 87 as i32,
         87 as i32, 93 as i32, 96 as i32];
    let mut param0_7: [i32; 26] =
        [80 as i32, 22 as i32, 38 as i32,
         26 as i32, 62 as i32, -(48 as i32),
         -(48 as i32), 46 as i32, -(54 as i32),
         4 as i32, 76 as i32, 46 as i32,
         48 as i32, 40 as i32, -(92 as i32),
         -(98 as i32), -(88 as i32), 12 as i32,
         -(42 as i32), -(94 as i32), 76 as i32,
         -(16 as i32), -(82 as i32), 62 as i32,
         18 as i32, -(24 as i32)];
    let mut param0_8: [i32; 14] =
        [0 as i32, 0 as i32, 0 as i32,
         0 as i32, 0 as i32, 0 as i32,
         0 as i32, 0 as i32, 1 as i32,
         1 as i32, 1 as i32, 1 as i32,
         1 as i32, 1 as i32];
    let mut param0_9: [i32; 49] =
        [85 as i32, 44 as i32, 1 as i32,
         97 as i32, 50 as i32, 74 as i32,
         62 as i32, 90 as i32, 3 as i32,
         78 as i32, 8 as i32, 48 as i32,
         96 as i32, 41 as i32, 36 as i32,
         91 as i32, 57 as i32, 97 as i32,
         85 as i32, 90 as i32, 78 as i32,
         43 as i32, 28 as i32, 92 as i32,
         85 as i32, 59 as i32, 29 as i32,
         38 as i32, 34 as i32, 65 as i32,
         20 as i32, 26 as i32, 27 as i32,
         23 as i32, 71 as i32, 22 as i32,
         47 as i32, 99 as i32, 68 as i32,
         93 as i32, 67 as i32, 66 as i32,
         69 as i32, 82 as i32, 98 as i32,
         15 as i32, 44 as i32, 51 as i32,
         65 as i32];
    let mut param0: [*mut i32; 10] =
        [param0_0.as_mut_ptr(), param0_1.as_mut_ptr(), param0_2.as_mut_ptr(),
         param0_3.as_mut_ptr(), param0_4.as_mut_ptr(), param0_5.as_mut_ptr(),
         param0_6.as_mut_ptr(), param0_7.as_mut_ptr(), param0_8.as_mut_ptr(),
         param0_9.as_mut_ptr()];
    let mut param1: [i32; 10] =
        [24 as i32, 6 as i32, 8 as i32,
         26 as i32, 8 as i32, 11 as i32,
         38 as i32, 22 as i32, 13 as i32,
         45 as i32];
    let mut i: i32 = 0 as i32;
    while i < param0.len() as i32 {
        if f_filled(param0[i as usize], param1[i as usize]) ==
               f_gold(param0[i as usize], param1[i as usize]) {
            n_success += 1 as i32
        }
        i += 1
    }
    println!("{} {} {} {} {}", "#Results:", " ", n_success, ", ", param0.len() as i32);
    return 0 as i32;
}
#[main]
pub fn main() { unsafe { ::std::process::exit(main_0() as i32) } }
